### DISCUSSION TITLE
Questions about scrolling a fixed size grid

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [zauberzeug/nicegui#2692] [Q&A] [answered]
### Question

I want to convert an existing gui to nicegui which will be accessed by consumers from any browser. The home page is a window spanning the screen containing a fixed-size grid (3 rows x 5 columns) of images, a text input box above the grid and buttons below the grid. Some questions:

a. Can the grid be scrolled up/down without scrollbars?

b. If there are a 1000 images to display, how are images fed or read into the grid - the infinite scroll example didn't help because mine is a fixed row-column grid?

c. Each image has a checkbox and a consumer can select multiple images. As they scroll, do checkbox selected images above and below the current view retain selected status when they come back into view?

**Comment by d-meehan** [comment]
I can attempt to answer a and c from working with scrollable areas.

a. Scroll area (https://quasar.dev/vue-components/scroll-area) let's you set styling for both the scroll bar and the thumb bar. By setting background:"transparent" for both using scroll_area().classes you should be able to make  the scroll bar invisible.

c. My understanding is that status should persist when an object isn't visible on the screen.

**Reply by dineshbvadhia** [reply]
Thanks for a) and c):

a) Using the example at https://nicegui.io/documentation/section_page_layout#scroll_area, how is background:"transparent" set?
```py
from nicegui import ui
with ui.row():
    with ui.scroll_area().classes('w-32 h-32 border'):
        ui.label('I scroll. ' * 20)
ui.run()
```
c) Agree, would expect state to be retained but will be good to receive confirmation

**Reply by dineshbvadhia** [reply]
Tried this but it didn't work:
```py
from nicegui import ui
with ui.row():
    with ui.scroll_area().classes('w-32 h-32 border background-transparent'):
        ui.label('I scroll. ' * 20)
ui.run()
```

**Comment by falkoschindler** [comment]
- a) Here is an example based on https://quasar.dev/vue-components/scroll-area#styled

    ```py
    with ui.scroll_area().classes('w-32 h-32 border') as area:
        area._props['thumb-style'] = {'background-color': 'transparent'}
        ui.label('I scroll. ' * 20)
    ```

- b) You can add all 1000 image elements to a scroll view. If you use `ui.image`, the images are not loaded immediately, but only when the image comes into view:

    ```py
    with ui.scroll_area().classes('w-32 h-32 border'):
        for i in range(100):
            ui.image(f'https://picsum.photos/id/{i}/640/360')
    ```

- c) Sure, the state is preserved.

**Reply by dineshbvadhia** [reply]
Thank-you :)

a) This makes the scroll bars transparent. Is it possible to scroll the scroll-area itself (in my case this is a fixed-size grid)?

**Reply by falkoschindler** [reply]
> Is it possible to scroll the scroll-area itself

Sorry, I don't understand.

**Reply by dineshbvadhia** [reply]
Running the example (on a Windows machine), you scroll on the left with the transparent scroll-bars. Can you instead scroll the viewport content (the list of 'I scroll')?
```py
with ui.scroll_area().classes('w-32 h-32 border') as area:
    area._props['thumb-style'] = {'background-color': 'transparent'}
    ui.label('I scroll. ' * 20)
```

**Reply by falkoschindler** [reply]
@dineshbvadhia You mean you want to move the mouse over the "I scroll" labels and then use the scroll wheel to scroll? Or do you want to "grab" the content and drag it to scroll it up and down?

Either way, this might be a browser-specific behavior I'm not too familiar with.

**Reply by dineshbvadhia** [reply]
Grab the content with a mouse depress, and then drag it up and down. It works in my current gui app (using kivy ScrollView). It is not browser-specific.

**Reply by falkoschindler** [reply, marked as answer]
Ok, I don't think this is possible with `ui.scroll_area`, since Quasar's Scroll Area doesn't support it and there's no easy workaround I'm aware of.
